sdk/test/opentelemetry/sdk/trace/span_test.rb

```diff
@@ -446,5 +446,13 @@
                       [], nil, links, Time.now, nil, nil)
       _(span.links.size).must_equal(1)
     end
+
+    it 'prunes invalid links' do
+      invalid_context = OpenTelemetry::Trace::SpanContext.new(trace_id: OpenTelemetry::Trace::INVALID_TRACE_ID)
+      links = [OpenTelemetry::Trace::Link.new(invalid_context)]
+      span = Span.new(context, Context.empty, 'name', SpanKind::INTERNAL, nil, span_limits,
+                      [], nil, links, Time.now, nil, nil)
+      _(span.links.size).must_equal(0)
```

Comment:
Consider using predicate assertions since it will print more details about the contents of the `span.links` array as part of the error message:

```suggestion
      _(span.links).must_be_empty
```

It may also be helpful, but not required, to include some details in the test to ensure the `links` parameter is not mutated per @fbogsany comment above. E.g. check that the links array is unchanged or use a frozen array as an argument.